### PR TITLE
dash/datagrid2-css-refactor

### PR DESCRIPTION
**Pull request by sebastianbochan**

### CODE REVIEW

**Comment by stitot** [comment]
Nice work, but some comments below. 

This should be placed further down with the theme CSS, and AFTER the selector where top-border is specified, for readability. It's an override, and should be positioned after the selector it overrides.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L106-L109

Is this needed?
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L111-L113

IMO we are somewhat inconsistent here. We use a class name to target the `tr`, but not the `td`. Yes, the class name has "row" in it, but it's not clear that this CSS selector applies to `tr`s in `tbody` only, and not `thead`.

To be more educational and clearer in our default CSS I would prefer:
`.highcharts-datagrid-table tbody tr` and `.highcharts-datagrid-table tbody td` as selectors here. This also paves the way for `.highcharts-datagrid-table tfoot td` when we get there, so that we are absolutely clear on what applies to what.

We should not be afraid to be a little verbose on selectors if that makes things clearer and easier to understand.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L145-L156

I don't like this hardcoded height, but that's another issue than this CSS refactoring I guess.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L147

I believe this should go down with the theme CSS
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L154

This should be moved down to the theme CSS, and  after the selector it is overriding. In addition: we are mixing `thead` and `tbody` stuff here. Should split those so that they are located together with other thead and tbody stuff. Better to repeat stuff, again for "educational" purposes.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L158-L162

All resizer stuff should be moved up, together with other header stuff.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L164

Is this still in use?
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L200

IMO we don't need this. If people want to they can fix styling themselves.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L212-L216

Sorting stuff should be moved up together with other header stuff:
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L230

Correct here would be "Default theme", since it covers both light and dark.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L255

Again, I don't like these hardcoded heights, but maybe a separate issue. If this height is needed for proper rendering/functionality then it should not be down here.
https://github.com/highcharts/highcharts/blob/8b1a8378b8de73ddbec3217389c2c967ad0a4b64/css/datagrid/datagrid.css#L276-L279

File: css/datagrid/datagrid.css [hunk 1]
```diff
@@ -256,42 +199,122 @@
...
+}
+
+/* Table header */
+.highcharts-datagrid-table thead {
+    background-color: var(--highcharts-neutral-color-5);
+    border-bottom: 1px solid var(--highcharts-neutral-color-20);
+}
+
+.highcharts-datagrid-table thead tr {
+    height: 32px;
```
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply css/datagrid/datagrid.css:264]: Currently it is needed, in the closes future we can replace it with dynamic calculation in a separate ticket.

[the remark this answers is not in the document]
Comment (stitot) [reply css/datagrid/datagrid.css:264]: OK, then it should be moved up to the Core CSS.

**Review by stitot — APPROVED** [review]
But please consider moving that 32px thing up to Core

File: css/datagrid/datagrid.css [hunk 2]
```diff
@@ -256,42 +199,122 @@
...
+    background-color: var(--highcharts-highlight-color-10);
+}
+
+.highcharts-datagrid-table tbody td .highcharts-datagrid-focused-cell input,
+.highcharts-datagrid-table tbody td .highcharts-datagrid-focused-cell input:focus {
+    border: 1px solid #0000ed;
+    background-color: #b7d4ff;
+}
+
+/* End Default Theme */
```
Comment (ddragula) [inline css/datagrid/datagrid.css:320]: Isn't the CSS linter blocking your commit? 😄 It seems there should be an empty newline at the end of the file.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Please check `data-grid/basic/cell-formatting`. Sth if off
image

**Review by karolkolodziej — APPROVED** [review]

**Review by ddragula — CHANGES_REQUESTED** [review]
`data-grid/basic/strict-row-heights` does not work correctly
image

**Review by ddragula — APPROVED** [review]

**Pull request merged**